opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -331,6 +340,22 @@ enum DataPointFlags {
   // Bits 2-31 are reserved for future use.
 }
 
+// GaugeSubType defines valid gauge "sub-types", which are used to describe
+// "special" kinds of gauges which have additional properties. These sub-types
+// generally apply to metrics which originate from protocols which support
+// these additional types, but are represented as gauges in OTLP.
+enum GaugeSubType {
+  // GAUGE_SUBTYPE_NONE is the default GaugeSubType, which represents a gauge
+  // that does not have a sub_type.
+  GAUGE_SUBTYPE_NONE = 0;
+
+  // GAUGE_SUBTYPE_UNKNOWN is a GaugeSubType which represents a metric for
+  // which the type is not known, such as a Prometheus metric without a TYPE
+  // comment. For example, the metric may be a monotonically-increasing counter,
+  // or histogram, but type information was not included at the source.
+  GAUGE_SUBTYPE_UNKNOWN = 1;
```

Comment:
What other values do we anticipate adding to this enum?

Do we anticipate expanding this concept to other point types?

Reply:
See https://github.com/open-telemetry/opentelemetry-specification/issues/3058#issuecomment-1616061089 for my analysis of other protocols with types that aren't fully represented in OTel.  For example, it might make sense (with proper justification) to add collectd's `absolute` type, although i'm not aware of anyone that has asked for it, and it seems to itself be a "legacy" type it inherited from a tool.

I think the concept could be expanded to sums as well, but needs its own justification.  See the linked comment for types that would potentially make sense as sum subtypes.